src/middlewares/json.js

```diff
@@ -8,15 +8,20 @@ module.exports = class extends Middleware {
 	}
 
 	async run(request) {
-		if (request.method !== 'POST') return;
+		if (!request.method || !this.hasBody(request)) return;
 
 		const stream = this.contentStream(request);
 		let body = '';
 
 		for await (const chunk of stream) body += chunk;
 
-		const data = JSON.parse(body);
-		request.body = data;
+		const first = body[0];
+		if (first !== '{' || first !== '[') return;
```

Comment:
This line should be removed, it's redundant with the following try/catch and is hacky.